### PR TITLE
COS integration: Issue remaining metrics

### PR DESCRIPTION
**Pull request by cbartz**
Applicable spec: [ISD-075](https://discourse.charmhub.io/t/specification-isd075-github-runner-cos-integration/12084)

### Overview

Issue the remaining metrics (except jog queuing time) mentioned in the spec and update the grafana dashboard.

image

### Rationale

To implement the COS integration as specified in the spec, the GitHub Runner charm needs to log all the metrics mentioned in the spec.

### Juju Events Changes

n/a

### Module Changes

- `metrics` : Add `RunnerStop` and `Reconciliation` events
- `runner_metrics`: Adapt code to emit `RunnerStop` event
- `runner_manager`: Issue the `Reconciliation` event

### Library Changes

n/a

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by arturo-seijas — DISMISSED** [review]
LGTM

File: src/runner_manager.py [hunk 1]
```diff
@@ -399,55 +506,19 @@ def reconcile(self, quantity: int, resources: VirtualMachineResources) -> int:
...
-                    logger.info("Removed runner: %s", runner.config.name)
-            else:
-                logger.info("There are no idle runners to remove.")
+            self._remove_runners(count=-delta, runners=online_runners)
         else:
             logger.info("No changes to number of runners needed.")
 
+        if self.config.charm_state.is_metrics_logging_available:
+            end_ts = time.time()
+            self._issue_reconciliation_metric(
```
Comment (amandahla) [inline src/runner_manager.py:517]: It's not very clear to me: What does the issue_reconciliation_metric do?

Comment (cbartz) [reply src/runner_manager.py:520]: The`ìssue_reconciliation_metric` method creates the reconciliation metric log entry, which looks like this:
```json
{
"event": "reconciliation",
"log_timestamp": 1694687032,
"flavor": "small",
"crashed_runners": 10,
"idle_runners": 12,
"duration": 600,
}
```

**Review by arturo-seijas — APPROVED** [review]

**Review by amandahla — APPROVED** [review]
LGTM

**Pull request merged**